datafusion/physical-expr/src/aggregate/bool_and_or.rs

```diff
@@ -219,23 +218,26 @@ impl PartialEq<dyn Any> for BoolAnd {
 
 #[derive(Debug)]
 struct BoolAndAccumulator {
-    bool_and: ScalarValue,
+    acc: Option<bool>,
 }
 
 impl BoolAndAccumulator {
     /// new bool_and accumulator
     pub fn try_new(data_type: &DataType) -> Result<Self> {
-        Ok(Self {
-            bool_and: ScalarValue::try_from(data_type)?,
-        })
+        assert_eq!(data_type, &DataType::Boolean);
+        Ok(Self { acc: None })
     }
```

Comment:
If you just assert the `DataType` to be one constant value, then you can also just remove the parameter and return `Self`.